File: medium/324/324.py
from ast import List
from audioop import reverse
import typing

class Solution:
    def wiggleSort(self, nums) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nums.sort(reverse = True)
        mid = len(nums)//2
        nums[::2], nums[1::2] = nums[mid:], nums[:mid]
        
    # wiggleSort([3,5,2,1,6,4])

p = Solution()
a = [1,2,2,3]
p.wiggleSort(a)
print(a)

File: medium/324/324.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function wiggleSort(nums: number[]): void {
  const arr = nums.slice()
  arr.sort((a, b) => a - b)
  const n = nums.length
  const x = Math.floor((n + 1) / 2)
  for (let i = 0, j = x - 1, k = n - 1; i < n; i += 2, j--, k--) {
    nums[i] = arr[j]
    if (i + 1 < n)
      nums[i + 1] = arr[k]
  }
}
wiggleSort([1, 2, 2, 3])